nixos/lib/test-driver/test_driver/machine.py

```diff
@@ -699,20 +699,22 @@ def check_file(_: Any) -> bool:
         with self.nested("waiting for file ‘{}‘".format(filename)):
             retry(check_file)
 
-    def wait_for_open_port(self, port: int) -> None:
+    def wait_for_open_port(self, port: int, addr: str = "localhost") -> None:
         def port_is_open(_: Any) -> bool:
-            status, _ = self.execute("nc -z localhost {}".format(port))
+            status, _ = self.execute("nc -z {} {}".format(addr, port))
             return status == 0
 
-        with self.nested("waiting for TCP port {}".format(port)):
+        with self.nested("waiting for TCP port {} on {}".format(port, addr)):
```

Comment:
The PR looks great overall.

While you are touching these strings, can you please also switch them over to the interpolated format `f"some string with a {var}"` for improved readability?

Reply:
I can do that in a follow-up PR. (I feel it's out of scope for the current topic.)

Reply:
https://github.com/NixOS/nixpkgs/pull/208278